### PR TITLE
iox-#1192 clamp history request

### LINKED ISSUE #1192: Prevent gap in samples when `historyRequest` > `queueCapacity`
## Required information

**Operating system:**
Ubuntu 20.04.3 LTS

**Compiler version:**
gcc 8.4.0

**Observed result or behaviour:**
If the `iceoptions` example is adapted with the following values, there is a gap when joining late.

Configuration:
```
publisherOptions.historyCapacity = 10U;
subscriberOptions.queueFullPolicy = iox::popo::QueueFullPolicy::BLOCK_PRODUCER;

subscriberOptions.queueCapacity = 1U;
publisherOptions.subscriberTooSlowPolicy = iox::popo::ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
```

Receiver output:
```
iox-cpp-subscriber-with-options got value: 4
iox-cpp-subscriber-with-options got value: 9
iox-cpp-subscriber-with-options got value: 10
iox-cpp-subscriber-with-options got value: 11
iox-cpp-subscriber-with-options got value: 12
iox-cpp-subscriber-with-options got value: 13
iox-cpp-subscriber-with-options got value: 14
iox-cpp-subscriber-with-options got value: 15
```

**Expected result or behaviour:**
Receiving the latest and greatest sample

Receiver output:
```
iox-cpp-subscriber-with-options got value: 8
iox-cpp-subscriber-with-options got value: 9
iox-cpp-subscriber-with-options got value: 10
iox-cpp-subscriber-with-options got value: 11
iox-cpp-subscriber-with-options got value: 12
iox-cpp-subscriber-with-options got value: 13
iox-cpp-subscriber-with-options got value: 14
iox-cpp-subscriber-with-options got value: 15
```

**Conditions where it occurred / Performed steps:**
Adapt `iceoptions` with the options above.

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer
* Clamps `historyRequest` to `queueCapacity`

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #1192

### CODE REVIEW

**Review by elfenpiff — APPROVED** [review]

File: iceoryx_posh/test/moduletests/test_posh_runtime.cpp [hunk 1]
```diff
@@ -517,6 +517,22 @@ TEST_F(PoshRuntime_test, GetMiddlewareSubscriberWithQueueCapacityZeroClampsQueue
     EXPECT_EQ(1U, subscriberPort->m_chunkReceiverData.m_queue.capacity());
 }
 
+TEST_F(PoshRuntime_test, GetMiddlewareSubscriberWithHistoryRequestLargerThanQueueCapacityClampsToQueueCapacity)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "77ca8d29-ffcb-4860-bf07-0af30b352e5c");
+    iox::popo::SubscriberOptions subscriberOptions;
+    constexpr uint64_t EXPECTED_HISTORY_REQUEST = 1U;
```
Comment (MatthiasKillat) [inline iceoryx_posh/test/moduletests/test_posh_runtime.cpp:524]: Maybe we should test this with a non-default queue capacity > 1?

**Comment by MatthiasKillat** [comment]
The functional change is ok, but I think the test should be more general.

File: iceoryx_posh/test/moduletests/test_posh_runtime.cpp (hunk 1 above)
Comment (mossmaurice) [reply iceoryx_posh/test/moduletests/test_posh_runtime.cpp:524]: The default `queueCapacity` is not 1 but `SubscriberChunkQueueData_t::MAX_CAPACITY`.

Comment (MatthiasKillat) [reply iceoryx_posh/test/moduletests/test_posh_runtime.cpp:524]: Yes, true. I still do not quite like the one and would prefer some other number.

**Review by MatthiasKillat — APPROVED** [review]

**Pull request merged**